src/sha1_hash.cpp

```diff
@@ -75,14 +75,12 @@ namespace libtorrent
 				continue;
 			}
 
-#if TORRENT_HAS_SSE
-#ifdef __GNUC__
+#if TORRENT_HAS_BUILTIN_CLZ
 			return ret + __builtin_clz(v);
-#else
+#elif TORRENT_HAS_SSE && defined _MSC_VER
```

Comment:
The name is misleading, the macro only test for essentially `x86` and `x86_64` and this days `msvc` has support for `arm` too. In fact, the redundant test is for `_MSC_VER`, I wrote it as it is for clarity.